.github/workflows/validate_and_publish_release.yml

```diff
@@ -8,13 +8,57 @@ on:
       release_id:
         description: 'Release id to publish'
         required: true
-        default: ''
-
-# TODO(jennik): Add more validation. https://github.com/google/iree/issues/6316
+      package_version:
+        description: 'Version of the package'
+        required: true
+      build_run_id:
+        description: 'Run ID for the build_package.yml workflow that triggered this workflow'
+        required: true
 
 jobs:
+  validate_packages:
+    name: "Validate packages"
+    # TODO(jennik): Look into testing windows and macos builds.
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Download packages
+        id: download_packages
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{secrets.WRITE_ACCESS_TOKEN}}
+          workflow: build_package.yml
+          run_id: ${{ github.event.inputs.build_run_id }}
+      - name: Display downloaded files
+        run: |
+          pwd
+          ls -R
+      - name: Set up python
+        id: set_up_python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install python packages
+        id: install_python_packages
+        run: |
+          python -m pip install -f file://$PWD/artifact/ iree-compiler-snapshot iree-jax-snapshot iree-runtime-snapshot iree-tools-tflite-snapshot iree-tools-tf-snapshot iree-tools-xla-snapshot
+      - name: Run tools with basic commands
+        id: run_tools_with_basic_commands
+        run: |
+          tar -xf artifact/iree-dist-snapshot-${{ github.event.inputs.package_version }}-linux-x86_64.tar.xz
+          cd bin
+          ./iree-benchmark-module --help
+          ./iree-benchmark-trace --help
+          ./iree-flatcc-cli --help
+          ./iree-opt --help
+          ./iree-run-mlir --help
+          ./iree-run-module --help
+          ./iree-run-trace --help
+          ./iree-tblgen --help
+          ./iree-translate --help
```

Comment:
The logs at https://github.com/not-jenni/iree/runs/3536908833?check_suite_focus=true could be easier to read with some changes here. (this is mostly GitHub's fault for having a silly way of displaying logs :P)

Notice that all commands are grouped up at the top and it doesn't actually show which tool corresponds to each section of help output:
![image](https://user-images.githubusercontent.com/4010439/132544661-5d67577e-6e42-44f8-9e36-260eb4b77510.png)

The raw logs at least show some information from the start of the tool help output:

![image](https://user-images.githubusercontent.com/4010439/132544925-0256d465-4b05-4c5f-bc86-3e666336ca3c.png)

How about adding a step for each tool instead of grouping them all into this one step? Then they would be displayed separately in GitHub's UI and it would be easy to see if one particular tool was missing or failed to load.

Reply:
Here is an updated run for what this will look like: https://github.com/not-jenni/iree/runs/3547504471?check_suite_focus=true